README.md

```diff
@@ -363,6 +363,8 @@ Options:
   --type    Which report type to generate (text | json | hist[buckets]).
             [default: text]
 
+  --hist    Histogram buckets, e.g.: '[0,1ms,10ms]'
```

Comment:
I would call this flag `--buckets` instead and not make the output of the actual histogram optional. It should always be included in both the JSON and text report, with the `--buckets` value determined automatically from the data set when not specified. This, however, would mean that the algorithm for automatic binning must be appropriate for large data sets (i.e. worse-case O(n)).

Reply:
Ok, I'll change it to --buckets (won't get to it today, but soon).

I'll do the change to enable the histogram by default using automatic bins in a follow up PR once this one is in.


> This, however, would mean that the algorithm for automatic binning must be appropriate for large data sets (i.e. worse-case O(n)).

Yes. It's already O(num of samples), and num of bucket is "small" (<=16 in practice, <= 64 technically) so that's ok too.

Your comment did make me realize I can just preallocate 64 buckets and remove the extension logic.

I wanted to benchmark the bucket search code, but with 16 buckets I expect linear search will be close to ideal.